### PR TITLE
Refactor: Change validateConfig Function into Pure Function

### PR DESCRIPTION
**Pull request by haricnugraha**
### what issue does this pr address

1. Change validateConfig function into pure function

### how did you implement / how did you fix it

Move all side effect to outside validateConfig function

### how **DID** you test

npm run test

### CODE REVIEW

**Review by armedi — APPROVED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -94,23 +94,41 @@ class Monika extends Command {
 
     // Read the config
     const file = flags.config
-    const config: Config = await parseConfig(file)
+    const config = await parseConfig(file)
+    const { notifications, probes } = config
+
+    // warn if config is empty
+    if ((notifications?.length ?? 0) === 0) {
+      const NO_NOTIFICATIONS_MESSAGE = `Notifications config has not been set. We will not be able to notify you when INCIDENT happened!\nPlease refer to Monika documentation for setting the notifications config at https://hyperjumptech.github.io/monika/guides/notifications.`
```
Comment (sapiderman) [inline src/index.ts:102]: Notifications has not been set. We will not be able to notify you when an INCIDENT occurs!\nPlease refer to the Monika documentations on how to configure notifications at https://hyperjumptech.github.io/monika/guides/notifications.`

**Review by sapiderman — COMMENTED** [review]
some minor grammar as reviewed.

File: src/index.ts (hunk 1 above)
Comment (haricnugraha) [reply src/index.ts]: Fixed.

**Comment by haricnugraha** [comment]
> some minor grammar as reviewed.

Fixed. Please recheck @sapiderman.

File: src/utils/looper.ts [hunk 2]
```diff
@@ -23,21 +23,58 @@
...
 
 const MILLISECONDS = 1000
+const DEFAULT_THRESHOLD = 5
+
+function sanitizeProbe(probe: Probe, index: number): Probe {
+  const { name, incidentThreshold, recoveryThreshold, alerts } = probe
+  probe.id = `${index}`
+
+  if (!name) {
+    probe.name = `monika_${probe.id}`
```
Comment (dennypradipta) [inline src/utils/looper.ts:38]: IIRC there is a key called `name` in probe. I think it should use the probe name first, if there isn't any, fallback to `monika_${probe.id}`

Comment (haricnugraha) [reply src/utils/looper.ts:38]: Please create issue for this, because i only move existing code, without add or change it's behavior or output.

**Review by nicnocquee — APPROVED** [review]

**Review by sapiderman — APPROVED** [review]
looks good!

**Pull request merged**